lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -185,6 +199,20 @@ mkJormungandrLayer mgr baseUrl = JormungandrLayer
                         (BlockId parentId)
                         (Just count)
                 left ErrGetDescendantsNetworkUnreachable <$> defaultHandler ctx x
+    , postSignedTx = \tx -> ExceptT $ do
+        let e0 = "?"
+        run (const () <$> cPostSignedTx tx) >>= \case
+            Left (FailureResponse e)
+                | responseStatusCode e == status400 -> do
+                    let msg = T.decodeUtf8 $ BL.toStrict $ responseBody e
+                    return $ Left $ ErrPostTxBadRequest msg
+            Left (FailureResponse e)
+                | responseStatusCode e == status500 && responseBody e == e0 -> do
+                    let msg = T.decodeUtf8 $ BL.toStrict $ responseBody e
+                    return $ Left $ ErrPostTxProtocolFailure msg
```

Comment:
It'll be best to do the "other way around". Don't try to do the error handling upfront, write tests and see what sort of error handling you need...